src/test/javascript/spec/component/assessment-shared/assessment-locks.component.spec.ts

```diff
@@ -31,24 +50,64 @@ describe('AssessmentLocksComponent', () => {
                 { provide: LocalStorageService, useClass: MockSyncStorage },
                 { provide: SessionStorageService, useClass: MockSyncStorage },
                 { provide: TranslateService, useClass: MockTranslateService },
+                { provide: textAssessmentService, useClass: TextAssessmentService },
+                { provide: programmingAssessmentService, useClass: ProgrammingAssessmentManualResultService },
+                { provide: fileUploadAssessmentService, useClass: FileUploadAssessmentService },
+                { provide: modelingAssessmentService, useClass: ModelingAssessmentService },
             ],
-        }).compileComponents();
+        })
+            .compileComponents()
+            .then(() => {
+                programmingAssessmentService = TestBed.inject(ProgrammingAssessmentManualResultService);
+                textAssessmentService = TestBed.inject(TextAssessmentService);
+                fileUploadAssessmentService = TestBed.inject(FileUploadAssessmentService);
+                modelingAssessmentService = TestBed.inject(ModelingAssessmentService);
+            });
     });
 
     beforeEach(() => {
         fixture = TestBed.createComponent(AssessmentLocksComponent);
         component = fixture.componentInstance;
         fixture.detectChanges();
     });
-
     it('should create', () => {
         expect(component).toBeTruthy();
     });
 
     it('should call getAllLockedSubmissions on init', () => {
         component.ngOnInit();
-
         expect(component.getAllLockedSubmissions).toHaveBeenCalled;
-        expect(component.submissions.length).toBe(0);
+    });
+
+    it('should release lock for programming exercise', () => {
+        const cancelAssessmentStub = jest.spyOn(programmingAssessmentService, 'cancelAssessment').mockReturnValue(of());
```

Comment:
Seems like these two lines are equal for all test cases. You could consider to move them into a ``beforeEach``call.

Reply:
the assesment stub is different in each test case, the first arguments differ.
Apllied it for the window stub